test/src/test/java/org/corfudb/integration/MergeTestListener.java

```diff
@@ -0,0 +1,204 @@
+        log.info("Processing updates in snapshot sync");
+        results.getEntries().forEach((schema, entries) -> {
+            entries.forEach(e -> {
+                Sample.StringKey key = (Sample.StringKey) e.getKey();
+                if (schema.getTableName().contains("Local_Table00")) {
+                    if (!(e.getOperation() == CorfuStreamEntry.OperationType.CLEAR) &&
+                            !(existingKeysInMergedTable.contains(key))) {
+                        try (TxnContext tx = corfuStoreSink.txn(namespace)){
+                            if (e.getOperation() == CorfuStreamEntry.OperationType.UPDATE) {
+                                tx.putRecord(tx.getTable(MERGED_TABLE_NAME), key,
+                                                SampleSchema.SampleMergedTable.newBuilder().setPayload(e.getPayload().toString().replaceAll("[^0-9]", "")).build(),
+                                         e.getMetadata());
+                                existingKeysInMergedTable.add(key);
+                            }
+                            else {
+                                tx.delete(tx.getTable(MERGED_TABLE_NAME), key);
+                            }
+                            tx.commit();
+                        }
+                    }
+                }
+                else {
+                    if ((e.getOperation() == CorfuStreamEntry.OperationType.UPDATE) &&
+                            !(existingKeysInMergedTable.contains(key))) {
+                        updates.add(e);
+                        addedKeys.put(key, e);
+                    }
+                    else if ((e.getOperation() == CorfuStreamEntry.OperationType.DELETE) &&
+                            !(existingKeysInMergedTable.contains(key))) {
+                        updates.add(e);
+                        deletedKeys.add(key);
+                    }
+                }
+            });
+
+        });
+    }
+
+    /**
+     * Write records to merged table, and process updates from replicated
+     * tables.
+     *
+     * @param results Entries received in a single transaction as part of a log entry sync
+     */
+    @Override
+    protected void processUpdatesInLogEntrySync(CorfuStreamEntries results) {
+        log.info("Processing updates in log entry sync");
+        results.getEntries().forEach((schema, entries) -> {
+            entries.forEach(e -> {
+                updates.add(e);
+                Sample.StringKey key = (Sample.StringKey) e.getKey();
+                // Add the records to idfwMergedTable.  Add to existingKeysInMergedTable
+                // Any other processing specific to the local table
+                existingKeysInMergedTable.add(key);
+                try (TxnContext tx = corfuStoreSink.txn(NAMESPACE)){
+                    tx.putRecord(tx.getTable(MERGED_TABLE_NAME), key,
+                            SampleSchema.SampleMergedTable.newBuilder().setPayload(e.getPayload().toString().replaceAll("[^0-9]", "")).build(),
```

Comment:
DELETE check is needed here right? otherwise you will hit NPE when trying to get payload

Reply:
Resolved as the comment above.